usr/lib/common/buffer.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * COPYRIGHT (c) International Business Machines Corp. 2022
+ *
+ * This program is provided under the terms of the Common Public License,
+ * version 1.0 (CPL-1.0). Any use, reproduction or distribution for this
+ * software constitutes recipient's acceptance of CPL-1.0 terms which can be
+ * found in the file LICENSE file or at
+ * https://opensource.org/licenses/cpl1.0.php
+ */
+#define _GNU_SOURCE
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <buffer.h>
+
+#define BUFFER_INIT_SIZE	(128ul)
+
+struct _buffer {
+    size_t b_size;
+    char *b;
+};
+
+p11_buffer_t *p11_buffer_new(void)
+{
```

Comment:
I'm sure this suits your needs, however for a general buffer implementation we could allow to pass an initial buffer size here. Might save a realloc in a 2nd step when it's already clear that the initial size is not enough.

Reply:
I would like to implement it only when it is needed.